spec/services/wallets/update_service_spec.rb

```diff
@@ -52,5 +52,186 @@
         expect(result.error.error_code).to eq('wallet_not_found')
       end
     end
+
+    context 'with recurring transaction rules' do
+      around { |test| lago_premium!(&test) }
+
+      let(:recurring_transaction_rule) { create(:recurring_transaction_rule, wallet:) }
+      let(:rules) do
+        [
+          {
+            rule_type: 'interval',
+            interval: 'weekly',
+            paid_credits: '105',
+            granted_credits: '105',
+          },
+        ]
+      end
+      let(:update_args) do
+        {
+          id: wallet.id,
+          name: 'new name',
+          expiration_at: DateTime.parse('2022-01-01 23:59:59'),
+          recurring_transaction_rules: rules,
+        }
+      end
+
+      before { recurring_transaction_rule }
+
+      it 'creates a new rule and removes the old one' do
+        result = update_service.update(wallet:, args: update_args)
+
+        aggregate_failures do
+          expect(result).to be_success
+
+          rule = result.wallet.reload.recurring_transaction_rules.first
+
+          expect(result.wallet.reload.recurring_transaction_rules.count).to eq(1)
+          expect(rule.id).not_to eq(recurring_transaction_rule.id)
+          expect(rule.rule_type).to eq('interval')
+          expect(rule.interval).to eq('weekly')
+          expect(rule.threshold_credits).to eq(0.0)
+          expect(rule.paid_credits).to eq(105.0)
+          expect(rule.granted_credits).to eq(105.0)
+        end
+      end
+
+      context 'edits the interval rule' do
+        let(:rules) do
+          [
+            {
+              id: recurring_transaction_rule.id,
+              rule_type: 'interval',
+              interval: 'weekly',
+              paid_credits: '105',
+              granted_credits: '105',
+            },
+          ]
+        end
+
+        it 'updates the rule' do
+          result = update_service.update(wallet:, args: update_args)
+
+          aggregate_failures do
+            expect(result).to be_success
+
+            rule = result.wallet.reload.recurring_transaction_rules.first
+
+            expect(result.wallet.reload.recurring_transaction_rules.count).to eq(1)
+            expect(rule.id).to eq(recurring_transaction_rule.id)
+            expect(rule.rule_type).to eq('interval')
+            expect(rule.interval).to eq('weekly')
+            expect(rule.threshold_credits).to eq(0.0)
+            expect(rule.paid_credits).to eq(105.0)
+            expect(rule.granted_credits).to eq(105.0)
+          end
+        end
+      end
+
+      context 'changes the rule into threshold one' do
```

Comment:
RSpec/ContextWording: Context description should match /^when\b/, /^with\b/, or /^without\b/. (https://rspec.rubystyle.guide/#context-descriptions, https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ContextWording)